### PR TITLE
fix(frost): clear stale TMA descriptor flags for short varlen tails

### LINKED ISSUE #1013: [Bug]: FROST KDA illegal memory access with a 1-token tail sequence on B300
### Description

FROST KDA fails with `cudaErrorIllegalAddress (700)` on B300 when the last variable-length sequence has one token. It reproduces on the **first forward**, without FLA, backward, or CUDA Graph.

Verified with a clean build (including the native extension) of `develop` at `68358074431265c84d8cc02ddeb528f929432596`, with no local kernel changes.

### Minimal reproducer

Save as `repro.py` and run `python3 repro.py`. Run each variant in a fresh process after an illegal access.

```python
import os
os.environ["CUDA_LAUNCH_BLOCKING"] = "1"
os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":4096:8"

import torch
import torch.nn.functional as F
from cudnn.linear_attention import kimi_delta_attention

torch.manual_seed(48)
torch.use_deterministic_algorithms(True)
T, H, D = 8192, 64, 128
q = F.normalize(torch.randn(T, H, D, device="cuda", dtype=torch.bfloat16), dim=-1)
k = F.normalize(torch.randn_like(q), dim=-1)
v = torch.randn_like(q)
g = torch.randn_like(q) * 0.1
beta = torch.ones(T, H, device="cuda", dtype=torch.bfloat16)
a_log = torch.zeros(H, device="cuda")
dt = torch.exp(torch.rand(H, D, device="cuda") * 4.605170186 - 6.907755279)
dt_bias = dt + torch.log(-torch.expm1(-dt))
cu = torch.tensor([0, 8191, 8192], device="cuda", dtype=torch.int32)

with torch.no_grad():
    out = kimi_delta_attention(
        q=q, k=k, v=v, g=g, beta=beta,
        a_log=a_log, dt_bias=dt_bias,
        safe_gate=True, gate_lower_bound=-5.0,
        cu_seqlens=cu, batch_invariant=True,
        checkpoint_every_n_tokens=0, plan_name="kda_frost",
    )[0]
torch.cuda.synchronize()
assert out.isfinite().all()
```

### Expected / actual

Expected: finite output for valid sequence boundaries.

| `cu_seqlens` | `batch_invariant=False` | `batch_invariant=True` |
| --- | --- | --- |
| `[0, 8192]` | Pass | Pass |
| `[0, 8191, 8192]` | Illegal access | Illegal access |

Relevant stack:

```text
frost/kda_engine.py:327 -> chunk_kda_sm100
frost/kernel/kda_prefill_f16.py:3275 -> compiled(...)
cutlass.cutlass_dsl.tvm_ffi_provider.CUDADialectError:
cudaErrorIllegalAddress (error code: 700)
```

### Environment

B300 (SM103), NVIDIA driver 580.105.08, Linux x86_64, Python 3.12, PyTorch 2.9.0 / CUDA 13.0,
cuDNN backend 9.24.0, cuDNN Frontend 1.29.0 at the commit above, CuTe DSL 4.7.0,
apache-tvm-ffi 0.1.11. No known working version; upgrading from TVM-FFI 0.1.10 did not resolve it.

### PR DESCRIPTION
**Pull request by Butterfingrz**
> **Credit: The idea for this fix comes entirely from [TMA Descriptor Encoding and the Hidden 21st Bit](https://zhuanlan.zhihu.com/p/2037200219700449995).**

## Before submitting

- [x] I agree to license this contribution under the terms of [LICENSE.txt](https://github.com/NVIDIA/cudnn-frontend/blob/develop/LICENSE.txt).
- [x] I ran `pre-commit run` and committed any formatting changes.
- [x] I reviewed the applicable Hard Rules in `AGENTS.md` and my changes comply, or I explain the exception below.
- [x] I added the required `cat-*`, `area:*` / `op:*`, and `orig-*` labels.

## Affected area

FE OSS kernels or CuTeDSL — FROST linear attention and shared THD descriptor helpers.

## Summary

Fix an illegal memory access in FROST KDA prefill when a packed variable-length input ends with a short sequence. The reported configuration uses BF16 inputs with `(T, H, D) = (8192, 64, 128)` and `cu_seqlens = [0, 8191, 8192]`; the first forward call fails with `cudaErrorIllegalAddress` before this fix.

Add `copy_tensor_map()` to both FROST THD modules. It clears bit 21 of the second 64-bit descriptor word while copying the 128-byte tensor map. Route `emit_seq_descs`, `emit_checkpoint_seq_descs`, and `emit_clamped_desc` through this helper before applying their descriptor updates.

## Why

Both figures below are from the blog cited above.

image

On the tested R580 driver, the encoder sets bit 21 when `product(dims) * element_size >= 128 KiB`.

A BF16 descriptor encoded for `(D, H, T) = (128, 64, 8192)` (128 MiB) keeps hidden bit 21 set after `tensormap.replace` rebases and shrinks it to `(128, 64, 1)` (16 KiB), so out-of-bounds loads can fault on unmapped memory.

image

B200/300 tests showed that completed loads still zero-filled out-of-bounds rows with bit 21 set; the diagram omits this behavior

**Fix:** Apply `qword[1] &= ~(1 << 21)` during device-side descriptor copies. [CUTLASS clears the same bit during host encoding](https://github.com/NVIDIA/cutlass/blob/cdcf8d86daa9b417840fd99875a1b1af685d389d/include/cute/atom/copy_traits_sm90_tma.hpp#L971), but that check cannot account for later device-side rebasing or shrinking.

## Related issues

Fixes [#1013](https://github.com/NVIDIA/cudnn-frontend/issues/1013).

## API and compatibility impact

No API or dependency changes. The fix only changes how FROST copies TMA descriptors.

## Testing

Tests passed on NVIDIA B200 (SM100, driver 580.126.09) and B300 (SM103, driver 580.95.05), both using PyTorch 2.13.0+cu130 (CUDA 13.0), cuDNN 9.20.0, and CuTeDSL 4.7.1.

- **Reproducer:** The original reproducer now passes. Additional checks passed with tail lengths of 4 at H=64, 2 and 3 at H=128, and 6 at H=32.
- **Numerics:** Packed outputs are bit-identical to separate per-sequence runs. Tail RMS error ratios against FP64 are `3.2e-3`–`3.6e-3`.
- **LA L0, non-exclusive:** 895 passed, 776 skipped, 2 xfailed, 0 failed.
- **LA hang-stress, GPU-exclusive:** 9 passed, 3 skipped, 0 failed.
- **SDPA THD subset:** 301 passed, 0 failed.
- **Performance:** Median latency changed by less than 0.1% across four tested shapes and three alternating A/B rounds. Differences were within run-to-run variation.

### CODE REVIEW

**Review by jhjpark — APPROVED** [review]

**Comment by jhjpark** [comment]
Thanks for helping to fix this issue!

**Comment by jhjpark** [comment]
@cudnn-ci-bot run frost
